crates/search/src/buffer_search.rs

```diff
@@ -2765,59 +2748,22 @@ mod tests {
             search_bar.deploy(&deploy, window, cx);
             assert_eq!(
                 search_bar.search_options,
-                SearchOptions::NONE,
-                "After hiding and showing the search bar, default options should be used"
+                SearchOptions::WHOLE_WORD,
+                "After hiding and showing the search bar, options should be preserved"
             );
 
             search_bar.toggle_search_option(SearchOptions::REGEX, window, cx);
             search_bar.toggle_search_option(SearchOptions::WHOLE_WORD, window, cx);
             assert_eq!(
                 search_bar.search_options,
-                SearchOptions::REGEX | SearchOptions::WHOLE_WORD,
+                SearchOptions::REGEX,
                 "Should enable the options toggled"
             );
             assert!(
                 !search_bar.dismissed,
                 "Search bar should be present and visible"
             );
         });
-
-        update_search_settings(
```

Comment:
That is actually an interesting case: what does happen to the search field now, if the settings are updated?
Would not it make sense to re-apply settings in this case?

Definitely worth having a test on that, one way or another.

Reply:
I think re-applying settings makes total sense as well, I'm slightly concerned though how it will feel since everything else about the search still persists except the `SearchOptions` specifically, but I think that's a pretty small detail and I'm not sure how much it actually matters